app/controllers/api/v0/highlights_controller.rb

```diff
@@ -37,14 +39,12 @@ class Api::V0::HighlightsController < Api::V0::BaseController
   end
 
   def create
-    binding, error = bind(params.require(:highlight), Api::V0::Bindings::NewHighlight)
+    inbound_binding, error = bind(params.require(:highlight), Api::V0::Bindings::NewHighlight)
     render(json: error, status: error.status_code) and return if error
 
-    # temporary test user_uuid.  will be removed when auth is set up
-    temp_user_uuid = '55783d49-7562-4576-a626-3b877557a21f'
-    created_highlight = Highlight.create!(binding.to_hash.merge(user_uuid: temp_user_uuid))
+    model = inbound_binding.create_model!(user_uuid: current_user_uuid)
```

Comment:
seems odd to me to put the model factory in the binding.  it's much more common to see different createX factories in the model. that's where a rails dev would first think of going to find a create factory for a model. was there a specific reason to put this into the binding?

Reply:
i do see one of your reasons...in the previous form, was pulling out the binding's innards (the hash) in order to pass thru to HL.create.   so, yeah...maybe this form is a bit better OO.    still have the bang? tho (referenced below)

Reply:
To me, the model is what other things depend on; it shouldn't have dependencies out to other things.  So I like that the bindings know about the model but not the other way around.  If the createX factory methods were in the model we'd eventually have `createFromV0Binding`, `createFromV1Binding`, etc, which I think you'd agree wouldn't be good.  For every binding version, we need a place to put code that goes from binding to model and model to binding.  The two options I see are putting that code in the binding or making a new mapper class for each binding (e.g. `Api::V0::BindingModelMappers::Highlight`).  I tend to prefer in the binding approach but I could go either way.  What do you think?

The bang is to reference that the model is being created with `create!` (to indicate that if errors happen they will raise exceptions instead of setting errors on the highlight model).

Reply:
"If you have a method that is marked with a bang at its end, then a version without it should also exist." https://blog.codeminer42.com/and-understanding-one-of-rubys-coolest-naming-conventions-5a9300b75605

Reply:
agree about the flow of dependency arrows.  but i wasnt arguing to put a dependency on the binding inside the model; i was suggesting a clean factory method inside the model that is passed the params.  or perhaps another abstraction.   that said..i think the binding factory is fine, if it's consistent.

Reply:
Yeah I know that some people have that opinion about `!` I just don't subscribe to it so wholeheartedly :-)